Handle empty size

pdx-daniel opened the pull request:
- Fixes #25 , where empty size strings produced invalid twemoji urls
- Adds a unit test for the above issue
- Adds a unit test for emoji 14 support

pdx-daniel committed:
add failing test

pdx-daniel committed:
update test to include extension

pdx-daniel committed:
add getSource to handle 'svg/' issue

pdx-daniel committed:
Merge branch 'master' into handle-empty-size

oztune commented on lib/makeTwemojiRenderer.js:
```diff
@@ -60,4 +53,25 @@ function merge() {
+
+	source += options.baseUrl
+
+	if(options.size === '' && options.baseUrl.endsWith('/') && options.ext === '.svg'){
```
I think adding svg to the url should be more explicit, and not an implicit behavior from passing `size: ''`. So I think a more robust solution for better svg support would require a larger API change. We're also very sensitive to the bundle size so I'd suggest a simpler fix of simply dropping the double slash with minimal modifications to line 36. This way our existing API would work more as expected, but would still require the users to modify the base url to add `/svg`.

Another consideration is: would making this change break the issue reporter's existing bug fix https://github.com/line/armeria/pull/4646/files#diff-2f901e80d43fc4fbadc238e158e1f4f161fbc9baf23d030dfa4e331a1b0e3b0c

oztune commented:
After some discussion we think the right solution is to:
- If a size wasn't provided by the user (either null or empty string), don't include the size component in the url at all (to avoid the `//`)
- If the base url doesn't have a trailing slash we'll insert it

pdx-daniel committed:
"If the base url doesn't have a trailing slash we'll insert it" - @oztune https://github.com/appfigures/react-easy-emoji/pull/26#issuecomment-1414226967

pdx-daniel committed:
Handle empty size
"If a size wasn't provided by the user (either null or empty string), don't include the size component in the url at all (to avoid the //)" - @oztune , https://github.com/appfigures/react-easy-emoji/pull/26#issuecomment-1414226967

pdx-daniel committed:
update interface comments to match implementation

oztune commented on lib/makeTwemojiRenderer.js:
```diff
@@ -60,4 +53,29 @@ function merge() {
+		source += normalizeProtocol(options.protocol)
+	}
+
+	var baseUrl = options.baseUrl.endsWith('/') ? options.baseUrl : options.baseUrl + '/'
```
We shouldn't use endsWith as it would break with older browsers

pdx-daniel replied:
https://github.com/appfigures/react-easy-emoji/pull/26/commits/480e477ead4cde8b3226d5bb1f3c8b497f3440a6

oztune commented on lib/makeTwemojiRenderer.js:
```diff
@@ -60,4 +53,29 @@ function merge() {
+	if(options.size === '' || options.size == null){
+		if(options.ext === '.svg'){
+			//Handle https://github.com/appfigures/react-easy-emoji/issues/25
+			source += 'svg/'
```
Instead of injecting this into the path, we should let the user manage it through the `baseUrl` param as the project README shows

pdx-daniel replied:
https://github.com/appfigures/react-easy-emoji/pull/26/commits/dec646eb76faaf4e3d0ba73546f647b4ee3c7856

oztune commented on lib/makeTwemojiRenderer.js:
```diff
@@ -60,4 +53,29 @@ function merge() {
    return out
+}
+
+function getSource(options, icon){
```
We should fold this function back into renderTwemoji, since the only change that needs to happen is on line 35

pdx-daniel replied:
https://github.com/appfigures/react-easy-emoji/pull/26/commits/167ab40e66d47faf4c5e1005c9488bc78ae765f6

pdx-daniel committed:
don't use endsWith https://github.com/appfigures/react-easy-emoji/pull/26#discussion_r1100467915

pdx-daniel committed:
manage svg through baseUrl https://github.com/appfigures/react-easy-emoji/pull/26#discussion_r1100468663

pdx-daniel committed:
fold getSource back into renderTwemoji https://github.com/appfigures/react-easy-emoji/pull/26/files#r1100469302
